ethcore/src/miner/miner.rs

```diff
@@ -411,12 +419,13 @@ impl Miner {
 			// | NOTE Code below requires transaction_queue and sealing_work locks.     |
 			// | Make sure to release the locks before calling that method.             |
 			// --------------------------------------------------------------------------
-			self.prepare_sealing(chain);
+			let (block, original_work_hash) = self.prepare_block(chain);
+			self.prepare_work(block, original_work_hash);
```

Comment:
```rust
#![feature(fn_traits, unboxed_closures)]

use std::ops::Fn;

fn main() {
    Fn::call(&foo, (1, 2.0));    
}

fn foo(x: i32, y: f32) {
    println!("({}, {})", x, y);
}
```

(although having to account for the implicit self makes that impossible in this case)